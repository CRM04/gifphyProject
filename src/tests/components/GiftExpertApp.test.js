import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GiftExpertApp from '../../components/GitfExpertApp';


describe('GiftExpertApp test suit', () => {
    
    test('GiftExpertApp should match snapshot', () => {
        const wrapper = shallow(<GiftExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('GiftExpertApp should show a list of categories', () => {
        const categories = ['Naruto', 'Goku'];
        const wrapper = shallow(<GiftExpertApp categories={ categories }/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GiftGrid').length).toBe(categories.length);
    });
    
});
