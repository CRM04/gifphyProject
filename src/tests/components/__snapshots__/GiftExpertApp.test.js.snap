// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftExpertApp test suit GiftExpertApp should match snapshot 1`] = `
<div>
  <h2>
     GitfExpertApp 
  </h2>
  <AddCategory
    setHeroes={[Function]}
  />
  <hr />
</div>
`;

exports[`GiftExpertApp test suit GiftExpertApp should show a list of categories 1`] = `
<div>
  <h2>
     GitfExpertApp 
  </h2>
  <AddCategory
    setHeroes={[Function]}
  />
  <hr />
  <GiftGrid
    heroe="Naruto"
    key="Naruto"
  />
  <GiftGrid
    heroe="Goku"
    key="Goku"
  />
</div>
`;
